web/ui/react-app/src/Graph.tsx

```diff
@@ -260,12 +291,27 @@ class Graph extends PureComponent<GraphProps> {
     if (this.props.data.result.length === 0) {
       return <Alert color="secondary">Empty query result</Alert>;
     }
-
+    const { selectedSeriesIndex } = this.state;
+    const series = this.getData();
+    const canUseHover = series.length > 1 && selectedSeriesIndex === -1;
     return (
       <div className="graph">
-        <ReactResizeDetector handleWidth onResize={() => this.plot()} />
+        <ReactResizeDetector handleWidth onResize={this.plot} />
         <div className="graph-chart" ref={this.chartRef} />
-        <Legend series={this.getData()} />
+        <div className="graph-legend" onMouseOut={canUseHover ? this.onLegendMouseOut : undefined}>
+          {series.map(({ index, color, labels }) => (
+            <div
+              style={{ opacity: selectedSeriesIndex > -1 && index !== selectedSeriesIndex ? 0.4 : 1 }}
+              onClick={series.length > 1 ? this.onSeriesSelect(index) : undefined}
```

Comment:
I believe renaming `onSeriesSelect` -> `handleSeriesSelect` would be more in line with React event handler naming conventions in general and in this codebase (same for `onSeriesHover`). The other convention I've seen is to omit the `on` / `handle` completely and just say `selectSeries`.

Reply:
Make sense, yes. This is why a style guide is needed :)